5.4.connected_vessels.py

```diff
@@ -0,0 +1,37 @@
+# The function accepts one or more iterable parameters, and returns a list of the interlaced elements
+def interleave(*iterable):
+    if not iterable:
+        return None
+    res = []
+    j = 0
+
+    # If we haven't reached the end of the first element
+    # - assuming that the size of all elements is the same
+    while j < len(iterable[0]):
+        for i in iterable:
```

Comment:
You could make this part of the code cleaner,  You should aim to write everything in a single line, even if you can. 
This link https://towardsdatascience.com/you-should-stop-craving-for-python-one-liners-5dbf8e0de23d explains it well, even though I disagree with how important memory and runtime are in most python programs.

Reply:
I fixed it.